[IMP] account, l10n_be: professional% column, private part account

ppat-odoo opened the pull request:
- account: When a Vendor Bill is posted with any account.move.line having Professional%(deductible_amount) other than 100(default), add a group group_purchase_all_invoicing_rights, which will toggle deductible_amount column's optional attribute from 'hide' to 'show'.

- l10n_be: For Purchase Journal add a default value in Private Part Account(non_deductible_account_id): '416000 Sundry Amounts Receivable'.

task-4661380

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/205491.png)](https://mergebot.odoo.com/odoo/odoo/pull/205491)

smetl reviewed (changes requested):
Quick review

ppat-odoo commented on addons/account/views/account_move_views.xml:
```diff
@@ -1188,7 +1188,13 @@
+                                               optional="hide"
+                                               groups="!account.group_purchase_all_invoicing_rights"/>
+                                        <field name="deductible_amount"
+                                               column_invisible="parent.move_type not in ['in_invoice', 'in_refund']"
```
We are adding this group so that in related enterprise PR when we predict the \`deductible_amount\`, that field is visible to user. But the predictions are done for only \`in_invoice\`, hence I am adding this group for only \`in_invoice\` and not for \`in_refund\`.

chklop commented on addons/account/security/account_security.xml:
```diff
@@ -93,6 +93,9 @@
     </record>
+    <record id="group_purchase_all_invoicing_rights" model="res.groups">
+        <field name="name">All Invoicing Rights</field>
+    </record>
```
Partial Purchase Deductibility

robodoo commented:
Merge method set to rebase and fast-forward.

ppat-odoo commented:
Hello @smetl :wave: ,
Staging failed due to changed import of `stepUtils` (ref: https://github.com/odoo/odoo/pull/217854) in tour.
I have fixed that and rebased the branches.
Can you please have a look at this PR.

robodoo commented:
Merge method set to rebase and fast-forward.
